### PR TITLE
implemented search with functionality

### PR DESCRIPTION
**Pull request by mxrcury**
implemented search with functionality

https://github.com/ita-social-projects/SpaceToStudy-Client/assets/90138904/7adf0678-115c-4453-bb86-108889e53a26

### CODE REVIEW

File: src/containers/my-resources/attachments-container/AttachmentsContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import SearchIcon from '@mui/icons-material/Search'
+
+import InputWithIcon from '~/components/input-with-icon/InputWithIcon'
+import { styles } from '~/containers/my-resources/attachments-container/AttachmentsContainer.styles'
+import { useDebounce } from '~/hooks/use-debounce'
+import useAxios from '~/hooks/use-axios'
+import { ResourceService } from '~/services/resource-service'
+
+import { Attachment, GetLessonsParams, ItemsWithCount, SortEnum } from '~/types'
+import { defaultResponses } from '~/constants'
```
Comment (Yurenko) [inline src/containers/my-resources/attachments-container/AttachmentsContainer.tsx:13]: Try to solve warnings in GitHub

Comment (Tolik170) [reply src/containers/my-resources/attachments-container/AttachmentsContainer.tsx:13]: Also, organize imports

Comment (mxrcury) [reply src/containers/my-resources/attachments-container/AttachmentsContainer.tsx:11]: ty, fixed

File: src/containers/my-resources/attachments-container/AttachmentsContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import InputWithIcon from '~/components/input-with-icon/InputWithIcon'
+import { useDebounce } from '~/hooks/use-debounce'
+import useAxios from '~/hooks/use-axios'
+import { ResourceService } from '~/services/resource-service'
+
+import { defaultResponses } from '~/constants'
+import { styles } from '~/containers/my-resources/attachments-container/AttachmentsContainer.styles'
+import { Attachment, GetLessonsParams, ItemsWithCount } from '~/types'
+
+const AttachmentsContainer: FC = () => {
```
Comment (Yurenko) [inline src/containers/my-resources/attachments-container/AttachmentsContainer.tsx:15]: FC is useless

Comment (mxrcury) [reply src/containers/my-resources/attachments-container/AttachmentsContainer.tsx]: removed

**Review by Yurenko — DISMISSED** [review]

**Review by Tolik170 — APPROVED** [review]

**Review by ArturBekhDEV — DISMISSED** [review]

**Review by maarekua — DISMISSED** [review]

**Review by Tolik170 — APPROVED** [review]

**Review by Mav-Ivan — APPROVED** [review]

**Review by Yurenko — APPROVED** [review]

**Pull request merged**